### ISSUE TITLE
Wrong column type for PLAN_HINT in statements summary

### ISSUE THREAD

**Issue by xhebox** [pingcap/tidb#56057]
## Bug Report

https://github.com/pingcap/tidb/blob/d6db1e8102b9c23a90e1e138dedcbcac89a6c389/pkg/infoschema/tables.go#L1393

It should be text, `PLAN_HINT` may be large. I encountered this when developing workload repository.

### 4. What is your TiDB version? (Required)

master-09-13

**Comment by tiancaiamao** [comment]
Do you mean `size: 64`  is too small for this column?
Maybe notice that enlarge value  might cause chunk size too big and increase the Go GC burden.
Ref https://github.com/pingcap/tidb/issues/28339

**Comment by xhebox** [comment]
> Do you mean `size: 64` is too small for this column? Maybe notice that enlarge value might cause chunk size too big and increase the Go GC burden. Ref #28339

Yeah, it is too small. There is no good workaround, since it will output much longer texts than the column definition. This works because that we don't check column types for internal mem table like this one.

**Comment by henrybw** [comment]
> This works because that we don't check column types for internal mem table like this one.

I think this is a deeper issue that needs to be addressed, as I suspect there are other bugs like this lurking in the codebase undetected.

**Comment by henrybw** [comment]
> > This works because that we don't check column types for internal mem table like this one.
> 
> I think this is a deeper issue that needs to be addressed, as I suspect there are other bugs like this lurking in the codebase undetected.

Filed a separate issue to track the more general problem: #60330.
